plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -73,6 +73,11 @@ type predicateMetadata struct {
 	podRequest                *schedulercache.Resource
 	podPorts                  map[int]bool
 	matchingAntiAffinityTerms []matchingPodAntiAffinityTerm
+
+	// CheckServiceAffinity: pods cache, otherwise this is a large query, will slow things down
+	// when there are 100s of services/pods.
+	lock        sync.Mutex
```

Comment:
Yes - findNodesThatFit will be called from scratch any time we will be scheduling a pod (including situations when e.g. pod failed to schedule and we are retrying). So the metadata will never be reused. So it's save to just precompute it.